src/c1_foeppl_von_karman/foeppl_von_karman_curvable_bell_elements.h

```diff
@@ -0,0 +1,1300 @@
+  class FoepplVonKarmanC1CurvableBellElement
+    : public virtual CurvableBellElement<NNODE_1D>,
+      public virtual FoepplVonKarmanEquations
+  {
+  public:
+    /// Constructor: Call constructors for C1CurvableBellElement and
+    /// FoepplVonKarman equations
+    FoepplVonKarmanC1CurvableBellElement()
+      : CurvableBellElement<NNODE_1D>(Nfield, Field_is_bell_interpolated),
+        FoepplVonKarmanEquations(),
+        Rotated_basis_fct_pt(0),
+        Nnodes_to_rotate(0)
+    {
+      // Use the higher order integration scheme
+      TGauss<2, 4>* new_integral_pt = new TGauss<2, 4>;
+      this->set_integration_scheme(new_integral_pt);
+
+      Association_matrix_pt = 0;
+    }
+
+    /// Destructor: clean up alloacations
+    ~FoepplVonKarmanC1CurvableBellElement() {}
+
+    /// Broken copy constructor
+    FoepplVonKarmanC1CurvableBellElement(
+      const FoepplVonKarmanC1CurvableBellElement<NNODE_1D>& dummy)
+    {
+      BrokenCopy::broken_copy("FoepplVonKarmanC1CurvableBellElement");
+    }
+
+    /// Broken assignment operator
+    void operator=(const FoepplVonKarmanC1CurvableBellElement<NNODE_1D>&)
+    {
+      BrokenCopy::broken_assign("FoepplVonKarmanC1CurvableBellElement");
+    }
+
+    /// Function pointer to basis vectors function which sets  basis vectors
+    /// b1 and b2 (which are in general functions of x)
+    typedef void (*BasisVectorsFctPt)(const Vector<double>& x,
+                                      Vector<double>& b1,
+                                      Vector<double>& b2,
+                                      DenseMatrix<double>& Db1,
+                                      DenseMatrix<double>& Db2);
+
+    /// Get the zeta coordinate
+    inline void interpolated_zeta(const Vector<double>& s,
+                                  Vector<double>& zeta) const
+    {
+      // If there is a macro element use it
+      if (this->Macro_elem_pt != 0)
+      {
+        this->get_x_from_macro_element(s, zeta);
+      }
+      // Otherwise interpolate zeta_nodal using the shape functions
+      else
+      {
+        interpolated_x(s, zeta);
+      }
+    }
+
+    /// Lagrange interpolated shape for in--plane displacements
+    void shape_u(const Vector<double>& s, Shape& psi) const;
+
+    /// Lagrange interpolated d_shape for in--plane displacements
+    void dshape_u_local(const Vector<double>& s,
+                        Shape& psi,
+                        DShape& dpsids) const;
+
+    /// Add the element's contribution to its residual vector (wrapper) with
+    /// cached association matrix
+    void fill_in_contribution_to_residuals(Vector<double>& residuals)
+    {
+      // Store the expensive-to-construct matrix
+      this->store_association_matrix();
+      // Call the generic routine with the flag set to 1
+      FoepplVonKarmanEquations::fill_in_contribution_to_residuals(residuals);
+      // Remove the expensive-to-construct matrix
+      this->delete_association_matrix();
```

Comment:
why store it and then delete it? I thought it would hang around for future calls?